packages/Php70/src/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector.php

```diff
@@ -0,0 +1,204 @@
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Broker\FunctionNotFoundException;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\ParametersAcceptor;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see https://www.php.net/manual/en/migration70.incompatible.php
+ *
+ * @see \Rector\Php70\Tests\Rector\FuncCall\NonVariableToVariableOnFunctionCallRector\NonVariableToVariableOnFunctionCallRectorTest
+ */
+final class NonVariableToVariableOnFunctionCallRector extends AbstractRector
+{
+    /**
+     * @var Broker
+     */
+    private $broker;
+
+    public function __construct(Broker $broker)
+    {
+        $this->broker = $broker;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('TO DO', [new CodeSample('reset(a());', '$a = a(); reset($a);')]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class, Node\Expr\MethodCall::class, Node\Expr\StaticCall::class];
+    }
+
+    /**
+     * @param FuncCall|Node\Expr\MethodCall|Node\Expr\StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $arguments = $this->getNonVariableArguments($node);
+
+        if ($arguments === []) {
+            return null;
+        }
+
+        foreach ($arguments as $key => $argument) {
+            $variable = null;
+            $assign = null;
+
+            if (
+                $argument instanceof Node\Expr\Assign
+                || $argument instanceof Node\Expr\AssignRef
+                || $argument instanceof Node\Expr\AssignOp
+            ) {
+                $variable = $argument->var;
+                $assign = $argument;
+            }
+
+            if (
+                $argument instanceof FuncCall
+                && $argument->name instanceof Node\Name
+            ) {
+                $name = $this->getName($argument);
+
+                if ($name === null) {
+                    continue;
+                }
+
+                $variable = new Variable($name);
+                $assign = new Node\Expr\Assign($variable, $argument);
+            }
+
+            if (
+                $argument instanceof Node\Expr\New_
+            ) {
+                if ($argument->class instanceof Node\Name) {
+                    $variable = new Variable(lcfirst($argument->class->getLast()));
+                } elseif ($argument->class instanceof Node\Stmt\Class_ && $argument->class->name !== null) {
+                    $variable = new Variable(lcfirst($argument->class->name->toString()));
+                } else {
+                    continue;
+                }
+
+                $assign = new Node\Expr\Assign($variable, $argument);
+            }
```

Comment:
I'm not sure what this section does.

It could be extrated into `processArgument()` or something similar, to improve readability.